### ISSUE TITLE
Add button processing state to forms that send data to server

### ISSUE THREAD

**Issue by jeffshaver** [mtr-cherish/cherish#130]
Disable button/change text while submission is happening so they are aware of the changes

**Comment by jeffshaver** [comment]
@iamlasse is this still an issue if we move the meteor methods out of the server folder in order to allow optimistic UI updates?

**Comment by iamlasse** [comment]
@jeffshaver I'm not sure i know what optimistic UI is..?

**Comment by chrisco23** [comment]
It's "optimistic" because it believes that its communication to the server will be handled appropriately, so presents the UI as if it had already happened.  http://info.meteor.com/blog/optimistic-ui-with-meteor-latency-compensation

**Comment by jeffshaver** [comment]
@iamlasse yeah. basically what chris said. So Meteor methods will run on the client and update minimongo (therefore updating the UI). The server has full final control of the outcome. But since the client can run the method, we get "optimistic" updates. Meaning that as long as the server agrees with the client about what happened, nothing changes from what the client did. 

This basically gets rid of server-round-trip time for methods to run by mimicking what the server will do. I have a ticket already to move the methods into the file with the collection they work against: #54. @ZachStoltz will be taking that one on I believe.

**Comment by weslers** [comment]
Would it not be better to keep the button disabled until all the form fields are filled in.

**Comment by jeffshaver** [comment]
@weslers This ticket is only related to button state while waiting for a server response. Form validation/disabling buttons until all fields are valid is handled in #129

**Comment by iamlasse** [comment]
I see, I will read more up on this.

**Comment by iamlasse** [comment]
@jeffshaver @wesler how does it work in blaze? In angular youd just set a ngdisabled conditional attr on buttons. Would it be a helper perhaps or an autorun?
